packages/client/wallets/aa/src/CrossmintAASDK.ts

```diff
@@ -1,151 +1,96 @@
-                        entryPoint
-                    );
-
-                    const abstractAddress = account.address;
-                    const { sessionKeySignerAddress } = await this.crossmintService.createSessionKey(abstractAddress);
-
-                    evmAAWallet.setSessionKeySignerAddress(sessionKeySignerAddress);
-
-                    await this.crossmintService.storeAbstractWallet({
-                        userIdentifier,
-                        type: ZERO_DEV_TYPE,
-                        smartContractWalletAddress: abstractAddress,
-                        eoaAddress: address,
-                        sessionKeySignerAddress,
-                        version: CURRENT_VERSION,
-                        baseLayer: "evm",
-                        chainId: blockchainToChainId(chain),
-                        entryPointVersion,
-                    });
-
-                    return evmAAWallet;
-                } catch (error: any) {
-                    throw new WalletSdkError(`Error creating the Wallet ${error?.message ? `: ${error.message}` : ""}`);
-                }
-            },
-            { user, chain }
-        );
-    }
-
     /**
-     * Clears all key material and state from device storage, related to all wallets stored. Call this method when the user signs out of your app, if you don't have a user identifier.
+     * Retrieves an existing wallet or creates a new one based on the provided parameters.
+     * - If a wallet does not exist for the given user and no `walletConfig` is provided, a passkey signer will be created by default.
+     * - If a wallet for a user already exists and `walletConfig` is provided, but the wallet and config do not match, an error will be thrown.
+     *
+     * @param {UserIdentifierParams} user - The user identifier parameters.
+     * @param {EVMBlockchainIncludingTestnet} chain - The blockchain type, including testnets.
+     * @param {WalletConfig} [walletConfig] - Optional configuration for the wallet, if specific wallet type needs to be created or validated.
+     * @returns {Promise<EVMAAWallet>} The wallet instance, either fetched or newly created.
      */
-    async purgeAllWalletData(): Promise<void> {
-        //Removes the Fireblocks NCW data stored on the localstorage
-        const keys = Object.keys(localStorage);
-        const keysToDelete = keys.filter((key) => key.startsWith("NCW-"));
-        keysToDelete.forEach((key) => {
-            localStorage.removeItem(key);
-        });
-    }
-
-    private async getEntryPointVersion<B extends EVMBlockchainIncludingTestnet = EVMBlockchainIncludingTestnet>(
-        userIdentifier: UserIdentifierParams,
-        chain: B | EVMBlockchainIncludingTestnet
-    ): Promise<EntryPointVersion> {
-        if (userIdentifier.email == null && userIdentifier.userId == null) {
-            throw new WalletSdkError(`Email or userId is required to get the entry point version`);
+    public async getOrCreateWallet(
```

Comment:
Here's the main proposed change for the SDK. Described above in comment

Reply:
Can we re-add the `logPerformance` wrapper?